runtime/doc/editorconfig.txt

```diff
@@ -13,16 +13,18 @@ the opened file are applied.
 
 For more information on EditorConfig, see https://editorconfig.org/.
 
-                                                       *g:editorconfig_enable*
-EditorConfig integration can be disabled by adding >lua
+                                               *g:editorconfig* *b:editorconfig*
+EditorConfig integration can be disabled globally by adding >lua
 
-    vim.g.editorconfig_enable = false
+    vim.g.editorconfig = false
 <
-to the user's |init.lua| file (or the Vimscript equivalent to |init.vim|).
+to the user's |init.lua| file (or the Vimscript equivalent to |init.vim|). It
+can also be disabled per-buffer by setting the |b:editorconfig| buffer-local
+variable to `false` or 0.
 
-                                                              *b:editorconfig*
 When Nvim finds a valid .editorconfig file it will store the applied
-properties in the buffer variable |b:editorconfig|.
+properties in the buffer variable |b:editorconfig| if it was not already set to
+`false` or 0 by the user.
```

Comment:
why support 0 also?

Reply:
That is vimscript `:help FALSE` (aka "falsey"), but why do we need it? vimscript has `v:false`.

Reply:
It was meant as a convenience. I don't see the harm in still supporting `0` as false, since many people have likely developed the habit of setting a variable to `0` to mean false. But we can omit that detail from the documentation.

Reply:
@justinmk sure, but then if you want to use a result of some expression in vim script, you have to do it like `let b:editorconfig = expr ? v:true : v:false`

Reply:
For establishing a convention, I think we should wait until there is a clear need for this. These kinds of conveniences increase surface area.

> but then if you want to use a result of some expression in vim script, you have to do it like `let b:editorconfig = expr ? v:true : v:false`

The vastly common case is a simple `let g:foo = v:false` to disable stuff. 

Reducing surface area allows us to not have to repeat extra boilerplate in every plugin that uses this convention.

Reply:
>For establishing a convention, I think we should wait until there is a clear need for this.

This convention is already well established though. `0` is false in Vimscript, and plenty of people still use `init.vim` for configuration and will write:

```vim
let g:editorconfig = 0
```

If we don't check for zero then this will not disable editorconfig, which is surprising and (arguably) a bug, since semantically that expression is "set `g:editorconfig` to false".

Reply:
we are establishing a new convention. people will paste into init.vim whatever we document, thus if we document:

    let g:editorconfig = v:false

then that's what will appear in init.vim files. Everywhere in the API we already require `v:false`, we don't do the "falsy" dance. Avoiding the "falsy" stuff avoids a lifetime of boilerplate and extra explanation in docs. 

Also, requiring `v:false` initially here, leaves the door open for changing our mind later and supporting "falsy". The converse is _not_ a two-way door, if we support falsy now we can't turn back.

This kind of "surface area" stuff adds permanent debt. That's why I am insisting on it.